### ISSUE TITLE
@ sync broken on julia 1.0

### ISSUE THREAD

**Issue by abannsunny** [JuliaWeb/Mux.jl#73]
on julia 1.0 even after defining @sync serve(app, 8000) julia just exits right away after the server is started. 

but if i run the script file in the REPL it doesn't exit and the server keeps listening to the defined port.

I switched to julia-0.6.2 and it works as expected. 

Thank you

**Comment by amgad-naiem** [comment]
@abannsunny This is fixed in https://github.com/JuliaWeb/Mux.jl/pull/71/files using `fetch(serve(app, port))`

**Comment by abannsunny** [comment]
Status `C:\Users\abann\.julia\environments\v1.0\Project.toml`
  [c43c736e] Genie v0.1.0 #master (https://github.com/essenciary/Genie.jl)
  [a975b10e] Mux v0.5.3
  [f269a46b] TimeZones v0.8.0

using fetch(serve(app, 7777))

still exits out. Am I missing something? this is my code.

using Mux

@app test = (
  Mux.defaults,
  page(respond("<h1>Hello World!</h1>")),
  page("/about",
       probabilty(0.9, respond("<h1>Boo!</h1>")),
       respond("<h1>About Me</h1>")),
  page("/user/:user", req -> "<h1>Hello, $(req[:params][:user])!</h1>"),
  Mux.notfound())

  println("server started")


  fetch(serve(test, 7777))

**Comment by emilyagras** [comment]
In the mean time, until the PR is closed, you can put `Base.JLOptions().isinteractive==0 && wait()` at the bottom of the program.  

This should work:

```
using Mux

@app test = (
  Mux.defaults,
  page(respond("<h1>Hello World!!!!</h1>")),
  page("/about",
       probabilty(0.1, respond("<h1>Boo!</h1>")),
       respond("<h1>About Me</h1>")),
  page("/user/:user", req -> "<h1>Hello, $(req[:params][:user])!</h1>"),
  Mux.notfound())

serve(test)
Base.JLOptions().isinteractive==0 && wait()
```

**Comment by jpsamaroo** [comment]
Fixed by #71 and #74 I believe.
